packages/gestalt/src/Checkbox/InternalCheckbox.tsx

```diff
@@ -180,10 +184,41 @@ const InternalCheckboxWithForwardRef = forwardRef<HTMLInputElement, Props>(funct
     [styles.readOnly]: readOnly,
   });
 
+  const iconWrapperStyles = classnames(styles.vrCheckIconWrapper, {
+    [styles.vrCheckIconEnterTransition]: !unchecked,
+    [styles.vrCheckIconExitTransition]: unchecked,
+  });
+
   return (
     <Box>
       <Box alignItems="start" display="flex" justifyContent="start" marginEnd={-1} marginStart={-1}>
         <Box paddingX={1} position="relative">
+          <div ref={tapScaleAnimation.elementRef} className={divStyles} style={style}>
```

Comment:
Moved custom checkbox element to the back of the input element. Now it's order in the DOM corresponds with the order in the stacking context. 

Because of the tap scale animation, this div now creates a new stacking context and was appearing on top of the input element (preventing it capturing mouse events). 
Read more [here](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context#:~:text=Element%20that%20has%20had%20stacking%20context%2Dcreating%20properties%20(such%20as%20opacity)%20animated%20using%20%40keyframes%2C%20with%20animation%2Dfill%2Dmode%20set%20to%20forwards.).